python/paddle/distributed/auto_parallel/operators/dist_default.py

```diff
@@ -0,0 +1,96 @@
+# Copyright (c) 2021 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License
+
+from .common import DistributedOperator
+from .common import DistributedOperatorImpl
+from .common import register_distributed_operator
+from .common import register_distributed_operator_impl
+from ..utils import is_dim_shard
+from ..utils import is_dim_replicate
+from ..utils import is_valid_list_index
+from ..utils import compute_compatible_dim_mapping
+from ..utils import compute_compatible_dims_mapping
+from ..utils import compute_compatible_and_update_dim_mapping
+from paddle.fluid import core, unique_name
+from paddle.fluid.framework import in_dygraph_mode
+from paddle.fluid.framework import Program, Parameter, Variable, program_guard
+from paddle.fluid.data_feeder import check_variable_and_dtype, check_dtype
+from ..process import new_process_group
+from ..utils import _get_comm_group
+
+
+class DistributedDefault(DistributedOperator):
+    def __init__(self, name):
+        super(DistributedDefault, self).__init__()
+        self._name = name
+
+
+register_distributed_operator("default", DistributedDefault("default"))
+
+
+# Replicated Default 
+class DistributedDefaultImpl0(DistributedOperatorImpl):
+    def __init__(self, name):
+        super(DistributedDefaultImpl0, self).__init__()
+        self._name = name
+        self._forward_implemented = True
+        self._backward_implemented = False
+
+    def is_process_mesh_compatible(self, op_dist_attr):
+        raise NotImplementedError("Please Implement this method.")
+
+    def is_input_compatible(self, op_dist_attr):
+        raise NotImplementedError("Please Implement this method.")
+
+    def is_output_compatible(self, op_dist_attr):
+        raise NotImplementedError("Please Implement this method.")
+
+    def update_dims_mapping(self, op_dist_attr):
+        raise NotImplementedError("Please Implement this method.")
+
+    @staticmethod
+    def forward(ctx, *args, **kwargs):
+
+        dst_block = ctx.get_dst_main_program().global_block()
```

Comment:
To get dst_block, we must first call get_dst_main_program. Can we directly to store the dst_block in the context? If we can, the dist op developers don't need to know the dst_main_program concept.